container/container.go

```diff
@@ -41,6 +42,7 @@ import (
 	"github.com/docker/libnetwork/netlabel"
 	"github.com/docker/libnetwork/options"
 	"github.com/docker/libnetwork/types"
+	agentexec "github.com/docker/swarmkit/agent/exec"
```

Comment:
`exec` is an interface package and these interfaces define how to interact with the executor and secrets. I'd be open to a better layout... it is not perfect.